File: backend/src/main/java/com/example/promotion/services/user/UserService.java
package com.example.promotion.services.user;

import com.example.promotion.dto.UserDTO;
import com.example.promotion.models.User;
import java.util.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.HashMap;


public interface UserService {

    User getUserById(long id);
    User createUser(User user);
    User updateUser(String email, User user);
    User findUser(String email);
    void deleteUser(String email);
    List<User> findAll();



}

File: frontend-pwa/vue/src/services/UserService.js
import authHeader from '../helpers/authHeader';
import axiosInstance from './AxiosInstance';

class UserService{

  createUser(user){
    return axiosInstance.post('/users', user)
  }

  getUserByEmail(email) {
    return axiosInstance.get('/users/' + email, { headers: authHeader() });
  }

  getUserById(id){
    return axiosInstance.get('/users/' + id)
  }

}

export default new UserService();